### DISCUSSION TITLE
How to request the status of a node

### DISCUSSION THREAD

**Discussion by PerRieland** [timmbogner/Farm-Data-Relay-System#130] [Q&A]
Thanks for a fantastic frame.
I am struggling to get the status of various "nodes". For example. I would like to request the status of a relay via MQTT. Is it ON or OFF?
These MQTT messages are defined in "fdrs_gateway_config.h"
// MQTT topics
#define TOPIC_DATA "fdrs/data"
#define TOPIC_STATUS "fdrs/status"
#define TOPIC_COMMAND "fdrs/command"
How to request the status of a node using MQTT?

**Comment by timmbogner** [comment]
Thanks, this is a really good question. This is a functionality that I should have built into the Irrigation sketch from the beginning.
The way I've been controlling relays is to simply send a DataReading object that is addressed to the relay's assigned ID, and with the data being a boolean representing on or off. The type is disregarded, mostly to simplify the code. 

  The proper scheme, which I'm laying out here with the intent of switching to, will require the user to specify the type of command: either set or get. 
  While sensors use globally-known reading types, controller types will be specific to the device. ~~I will use type "0" for "get", and "1" for "set"~~ . Once added, you will send a get request to like this:  ~~```[{"id":42,"type":0,"data":1}]``` (The "data" value of "1" will specifically ask for the relay's status~~.  In the future, I may want to add another parameter to get the device's battery voltage, and I may use "0" to ask for this.)
  The response will actually be very similar. 0 is the definition for the STATUS_T sensor type, so the relay will send the following to indicate the status of the relay: ```[{"id":42,"type":0,"data":0}]``` . 

I spent a while brainstorming and that's what I came up with. Let me know if it makes sense or any ideas you may have to improve this scheme. Thanks again!

**Reply by timmbogner** [reply]
UPDATE: I implemented it with [4c411a0](https://github.com/timmbogner/Farm-Data-Relay-System/commit/4c411a053ee10a32c29ed61b989a5b7b09193265).

In the end I used type 0 for SET and 1 for GET, so as to retain some backwards compatibility. So you query a node by sending ```[{"id":101,"type":1,"data":0}]```. I also omitted the data-value idea for now.
